File: backend/src/main/java/com/shoppingmall/entity/Product.java
package com.shoppingmall.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 商品实体类
 *
 * @author ShoppingMall Team
 * @date 2025-12-06
 */
@Data
@TableName("product")
public class Product {

    /**
     * 主键ID
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 商品编码/SKU（唯一）
     */
    private String productCode;

    /**
     * 商品名称
     */
    private String productName;

    /**
     * 分类ID
     */
    private Long categoryId;

    /**
     * 品牌ID（关联brand表）
     */
    private Long brandId;

    /**
     * 主图URL
     */
    private String mainImage;

    /**
     * 商品图片（JSON数组）
     */
    private String images;

    /**
     * 商品描述
     */
    private String description;

    /**
     * 销售价格
     */
    private BigDecimal basePrice;

    /**
     * 销售价格
     */
    private BigDecimal salePrice;

    /**
     * 建议零售价
     */
    private BigDecimal suggestedRetailPrice;

    /**
     * 市场零售价
     */
    private BigDecimal marketRetailPrice;

    /**
     * 会员价（启用时作为售价）
     */
    private BigDecimal memberPrice;

    /**
     * 是否启用会员价（0-否，1-是）
     */
    private Integer enableMemberPrice;

    /**
     * 库存数量
     */
    private Integer stock;

    /**
     * 警戒库存
     */
    private Integer warningStock;

    /**
     * 商品重量(g)
     */
    private Integer weight;

    /**
     * 销量
     */
    private Integer salesCount;

    /**
     * 状态（0-下架，1-上架）
     */
    private Integer status;

    /**
     * 逻辑删除（0-未删除，1-已删除）
     */
    @TableLogic
    private Integer deleted;

    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: admin-frontend/src/api/admin/product.ts
/**
 * 商品管理API
 */

import request from '@/utils/request'

// 商品DTO
export interface ProductDTO {
  id?: number
  productCode: string
  barcode?: string
  unit?: string
  productName: string
  categoryId: number
  brandId?: number | null
  shippingTemplateId?: number | null
  mainImage?: string
  images?: string
  description?: string
  basePrice: number
  suggestedRetailPrice?: number
  marketRetailPrice?: number
  memberPrice?: number
  enableMemberPrice?: number
  stock: number
  warningStock?: number
  weight?: number
  status?: string
  enableSpec?: boolean
}

// 商品VO
export interface ProductVO {
  id: number
  productCode: string
  barcode?: string
  unit?: string
  productName: string
  categoryId: number
  categoryName: string
  brandId?: number | null
  brandName?: string
  shippingTemplateId?: number | null
  shippingTemplateName?: string
  mainImage: string
  imageList: string[]
  description: string
  basePrice: number
  suggestedRetailPrice?: number
  marketRetailPrice?: number
  memberPrice?: number
  enableMemberPrice?: number
  userLevelPrice: number
  stock: number
  warningStock?: number
  weight?: number
  salesCount: number
  status: string
  createTime: string
  updateTime: string
}

// 分页响应
export interface PageResponse<T> {
  records: T[]
  total: number
  size: number
  current: number
  pages: number
}

/**
 * 分页查询商品列表
 */
export const getProductPage = (
  current: number,
  size: number,
  categoryId?: number,
  keyword?: string,
  brand?: string,
  status?: string,
  sortBy?: string
): Promise<PageResponse<ProductVO>> => {
  return request.get('/api/admin/product/page', {
    params: { current, size, categoryId, keyword, brand, status, sortBy }
  })
}

/**
 * 根据ID获取商品详情
 */
export const getProductById = (id: number): Promise<ProductVO> => {
  return request.get(`/api/admin/product/${id}`)
}

/**
 * 创建商品
 */
export const createProduct = (data: ProductDTO): Promise<number> => {
  return request.post('/api/admin/product', data)
}

/**
 * 更新商品
 */
export const updateProduct = (data: ProductDTO): Promise<void> => {
  return request.put('/api/admin/product', data)
}

/**
 * 删除商品
 */
export const deleteProduct = (id: number): Promise<void> => {
  return request.delete(`/api/admin/product/${id}`)
}

/**
 * 更新商品状态
 */
export const updateProductStatus = (id: number, status: string): Promise<void> => {
  return request.put(`/api/admin/product/${id}/status`, null, { params: { status } })
}

export interface ProductImportResult {
  totalCount: number
  successCount: number
  failCount: number
  errors: Array<{
    rowNumber: number
    productCode: string
    errorMessage: string
  }>
  warnings: string[]
}

export const importProducts = (csvFile: File, imageZip?: File): Promise<ProductImportResult> => {
  const formData = new FormData()
  formData.append('csvFile', csvFile)
  if (imageZip) {
    formData.append('imageZip', imageZip)
  }
  
  return request.post('/api/admin/product/import', formData, {
    headers: {
      'Content-Type': 'multipart/form-data'
    }
  })
}
